packages/webapp/pages/404.tsx

```diff
@@ -1,8 +1,24 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useContext, useEffect, useState } from 'react';
 import Custom404 from '@dailydotdev/shared/src/components/Custom404';
 import { NextSeo } from 'next-seo';
+import { AnalyticsEvent } from '@dailydotdev/shared/src/lib/analytics';
+import AnalyticsContext from '@dailydotdev/shared/src/contexts/AnalyticsContext';
 
 export default function Custom404Seo(): ReactElement {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [trackedImpression, setTrackedImpression] = useState(false);
+
+  useEffect(() => {
+    if (trackedImpression) {
+      return;
+    }
+
+    trackEvent({
+      event_name: AnalyticsEvent.View404Page,
+    });
+    setTrackedImpression(true);
+  }, [trackEvent, trackedImpression]);
```

Comment:
Can't we do this just on mount? Or trackEvent changes somehow?

Reply:
Mainly to avoid re-render triggers we always do the page view ones like this :)

Reply:
🤷‍♀️ yeah I've seen it a lot, not 100% ok with it personally in the idea, don't see why it would re-render, feels like either some extra unnecessary layer/code added, or we just haven't written our hook well.

But yeah ok with it for this, we can recheck the whole usage of these in a separate refactor if/when time comes.

Reply:
We don't have a hook is the problem :) 
It's a backlog ticket somewhere to have a custom once off tracking hook, but think we waiting for React 18 hooks :")

Reply:
> don't see why it would re-render

it is because `trackEvent` does not have stable reference but we need it as dependency when in `useEffect`, so to make sure we are really only sending one event per component lifecycle we have this hack. The new hook [useEventEffect](https://react.dev/learn/separating-events-from-effects#declaring-an-effect-event) in future React allows to create handlers that have stable reference and are not required as dependencies.